jib-maven-plugin/src/test/java/com/google/cloud/tools/jib/maven/MavenProjectPropertiesTest.java

```diff
@@ -589,32 +532,22 @@ public void testCreateContainerBuilder_warNonDefaultAppRoot()
             "/my/app/WEB-INF/classes/package",
             "/my/app/WEB-INF/classes/package/test.properties",
             "/my/app/WEB-INF/lib",
-            "/my/app/WEB-INF/web.xml"),
-        layers.resourcesLayers.get(0).getEntries());
-    assertExtractionPathsUnordered(
-        Arrays.asList(
+            "/my/app/WEB-INF/web.xml");
+    assertThat(layers.classesLayer.getEntries())
+        .comparingElementsUsing(EXTRACTION_PATH_OF)
+        .containsExactly(
             "/my/app/WEB-INF/classes/HelloWorld.class",
             "/my/app/WEB-INF/classes/empty_dir",
             "/my/app/WEB-INF/classes/package",
-            "/my/app/WEB-INF/classes/package/Other.class"),
-        layers.classesLayers.get(0).getEntries());
-  }
-
-  @Test
-  public void testCreateContainerBuilder_jarNonDefaultAppRoot()
-      throws IOException, InvalidImageReferenceException, CacheDirectoryCreationException {
-    // Test when the default packaging is set
-    Mockito.when(mockMavenProject.getPackaging()).thenReturn("jar");
-    BuildContext buildContext = setUpBuildContext("/my/app", DEFAULT_CONTAINERIZING_MODE);
-    assertNonDefaultAppRoot(buildContext);
+            "/my/app/WEB-INF/classes/package/Other.class");
   }
 
   @Test
   public void testCreateContainerBuilder_noErrorIfWebInfDoesNotExist()
       throws IOException, InvalidImageReferenceException, CacheDirectoryCreationException {
     setUpWar(temporaryFolder.newFolder("final-name").toPath());
 
-    setUpBuildContext("/anything", DEFAULT_CONTAINERIZING_MODE); // should pass
+    setUpBuildContext(); // should pass
```

Comment:
Hm can we add a few more assertions on what `setUpBuildContext` results in? It might help explain the behavior a bit better.

Reply:
I left a comment in the other PR: https://github.com/GoogleContainerTools/jib/pull/3112#discussion_r588701903